docs/self-hosted/kubernetes-helm.md

```diff
@@ -0,0 +1,106 @@
+
+Please note that if you choose to use the built-in solution then each of the storage requires one or more volumes using
+[dynamic volume provision](https://kubernetes.io/docs/concepts/storage/dynamic-provisioning/) to allocate storage for the different Pods.
+Configure the Storage Class in the values.yaml under `global.storageClass`.
+
+### Annotated configuration
+
+To install the Private Packagist Self-Hosted Helm Chart configure values based on your setup, store them in a YAML file, e.g. `values.yaml`, and then run the commands below.
+
+HELM_CHART_VALUES_FILE
+
+#### Login to the Helm registry and install the chart
+
+Make sure you replace `YOUR_USERNAME`, `YOUR_PASSWORD`, `values.yaml`, and `VERSION` with your values before running the commands.
+You can find the latest Private Packagist version on our [changelog page](/docs/self-hosted/changelog).
+
+```
+helm registry login registry.replicated.com --username YOUR_USERNAME --password YOUR_PASSWORD
+helm install -f values.yaml private-packagist oci://registry.replicated.com/privatepackagistkots/private-packagist --version VERSION
+```
+
+Verify that the helm chart is deployed by running `kubectl get pods` and verifying that the `repo`, `ui`, and `worker` pods are running.
+You should then be able to access the web interface by opening the configured domain in the browser, e.g. https://packagist.myintranet.com.
+
+#### Authentication Setup
+Within Private Packagist Self-Hosted, you now need to set up at least one user authentication method.
+You have the choice between authentication with email addresses and passwords and several OAuth integrations with third-party services.
+We provide integrations with on-premises versions of GitHub, Bitbucket, GitLab, or their public services on github.com, bitbucket.org,
+or gitlab.com. Follow the instructions to create the respective required identifiers, tokens, and secrets.
+
+* [GitHub (Enterprise) Integration Setup](./github-integration-setup.md)
+* [Bitbucket.org Integration Setup](./bitbucket-integration-setup.md)
+* [Bitbucket Data Center / Server Integration Setup](./bitbucket-server-integration-setup.md)
+* [GitLab Integration Setup](./gitlab-integration-setup.md)
+* [Authentication with Email Addresses and Passwords](./authentication-email-addresses-passwords-setup.md).
+
+
+![Add Integration](/Resources/public/img/docs/self-hosted/08-integration.png)
+
+#### Selecting Admins
+After setting up an integration, you can log in through the top menu. Register an account and pick a username.
+
+![Register Admin](/Resources/public/img/docs/self-hosted/09-register-admin.png)
+
+The first user is granted admin permissions automatically. You can grant admin permissions to more users in the admin panel.
+
+![Add Admin](/Resources/public/img/docs/self-hosted/10-add-admin.png)
+
+#### Switching to Production Mode
+Head back to the admin console to disable the Setup Mode in the configuration. Once the preflight checks passed, you can
+apply the changes.
```

Comment:
This stil needs to be updated/changed, there is no admin console. 

This is how I do it right now:

1. Change `application.setupMode.enabled` in values.yaml
2. Run an upgrade using the same version:  `helm upgrade private-packagist oci://registry.replicated.com/privatepackagistkots/private-packagist --version 2.0.5-a7 -f ~/values.yaml`
3. Restart the deployment to pick up the changes `kubectl rollout restart deployment ui` 

Or alternatively, by editing the configmap directly with `kubectl edit configmaps ui-parameters` but that's quite messy because parameters.yaml is one long string. 

Any other ideas for this?

Reply:
Thanks, will update and add a note about disabling the setupMode. What you describe seem to be the way to go. We could make the setupMode part of the UI definition, similar to how this is done with the sequence. Then this would automatically restrat.

Reply:
Updated